### PR TITLE
Add DurationPredicates

### PR DESCRIPTION
**Pull request by mikezaccardo**
Adds `DurationPredicates` which contains `Predicate`s for common `Duration` comparison operations.

### CODE REVIEW

File: utils/common/src/main/java/org/apache/brooklyn/util/time/DurationPredicates.java [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+     *            the returned {@link Predicate}.
+     * @return A {@link Predicate} that checks if the {@link Duration} that was
+     *         supplied to this method is longer than any {@link Stopwatch}
+     *         supplied to the returned {@link Predicate#apply(Object)}.
+     */
+    public static Predicate<Stopwatch> isLongerThanStopwatch(final Duration duration) {
+        return new Predicate<Stopwatch>() {
+            @Override
+            public boolean apply(Stopwatch other) {
+                return duration != null && other != null && duration.isLongerThan(other);
```
Comment (grkvlt) [inline utils/common/src/main/java/org/apache/brooklyn/util/time/DurationPredicates.java]: This is confusing. A `Predicate<Stopwatch>` is making a boolean statement about `Stopwatch`-es, so in this case maybe reverse the check and the name?

File: utils/common/src/main/java/org/apache/brooklyn/util/time/DurationPredicates.java [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+     *            the returned {@link Predicate}.
+     * @return A {@link Predicate} that checks if the {@link Duration} that was
+     *         supplied to this method is shorter than any {@link Duration}
+     *         supplied to to the returned {@link Predicate#apply(Object)}.
+     */
+    public static Predicate<Duration> isShorterThanDuration(final Duration duration) {
+        return new Predicate<Duration>() {
+            @Override
+            public boolean apply(Duration other) {
+                return duration != null && other != null && duration.isShorterThan(other);
```
Comment (grkvlt) [inline utils/common/src/main/java/org/apache/brooklyn/util/time/DurationPredicates.java]: Again, the `Duration` the predicate speaks about here is the one called `other`, the `duration` is constant over the life of the predicate.

File: utils/common/src/main/java/org/apache/brooklyn/util/time/DurationPredicates.java [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+     *            the returned {@link Predicate}.
+     * @return A {@link Predicate} that checks if the {@link Duration} that was
+     *         supplied to this method is shorter than any {@link Stopwatch}
+     *         supplied to to the returned {@link Predicate#apply(Object)}.
+     */
+    public static Predicate<Stopwatch> isShorterThanStopwatch(final Duration duration) {
+        return new Predicate<Stopwatch>() {
+            @Override
+            public boolean apply(Stopwatch other) {
+                return duration != null && other != null && duration.isShorterThan(other);
```
Comment (grkvlt) [inline utils/common/src/main/java/org/apache/brooklyn/util/time/DurationPredicates.java]: Same.

**Comment by grkvlt** [comment]
@mikezaccardo remember we need tests for these too!

**Comment by ahgittin** [comment]
LGTM

**Pull request merged**